Document custom image transformations (#620)
PyTorchLightning/lightning-flash: docs/source/reference/image_classification.rst

    :language: python
    :lines: 14-

> 

Reviewer: I think we could have a new section for this. Don't mind what you call it, `Custom Transformations` or something like that?

Author: Yes, `Custom Transformations` seems good

Flash automatically takes care of Image transformations for you. But sometimes it is not enough and you want to customize the transformations.
You can easily customize `Preprocessing <https://github.com/PyTorchLightning/lightning-flash/blob/master/flash/core/data/process.py>`_ method in the `data_module`.
The base `Preprocess` class defines 7 methods for different stages in the data loading pipeline.
To apply image augmentations you can directly use `default_transforms` defined in Flash Image module and then add your merge your custom defined image transformations with it.

> In this example, we will load the default transforms and merge it with custom `torchvision` transformations. We use `post_tensor_transform` since the transformations will be applied
> after image is converted into `torch.Tensor`.

Reviewer: ```suggestion
In this example, we will load the default transforms and merge with custom `torchvision` transformations. We use the `post_tensor_transform` hook to apply the transformations
after the image is converted to a `torch.Tensor`.
```

.. literalinclude:: ../../../flash_examples/image_transforms.py

>     :language: python

Reviewer: Let's use testcode here and have the example directly in the docs.

Author: didn't get what you mean by "Let's use testcode here" 😅. Could you pls explain this?
